### PR TITLE
TAI64 library for std

### PR DESCRIPTION
**Pull request by kristate**
Implements TAI64N

https://cr.yp.to/libtai/tai64.html

### CODE REVIEW

File: std/tai64.zig [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    // external TAI64 format, consisting of eight 8-bit bytes in big-endian
+    // format. This means that bytes b0 b1 b2 b3 b4 b5 b6 b7 represent the
+    // label b0 * 2^56 + b1 * 2^48 + b2 * 2^40 + b3 * 2^32 + b4 * 2^24 + b5 *
+    // 2^16 + b6 * 2^8 + b7.
+
+    // A TAI64N label refers to a particular nanosecond of real time.
+    // It has two parts: 1) a TAI64 label and 2) an integer, between 0
+    // and 999999999 inclusive, counting nanoseconds from the
+    // beginning of the second represented by the TAI64 label.
+    tai64n: [12]u8,
```
Comment (andrewrk) [inline std/tai64.zig]: Why store this encoded? Seems like this should be:

```zig
sec: u64,
nsec: u30,
```

`to_bytes` can be:

```zig
pub fn encode(self: TAI64N) [12]u8 {
    var result: [12]u8 = undefined;
    mem.writeInt(result[0..8], self.sec, builtin.Endian.Big);
    mem.writeInt(result[8..12], self.nsec, builtin.Endian.Big);
    return result;
}
```

Then we can have a corresponding `decode` function.

Comment (andrewrk) [reply std/tai64.zig]: Forgot to add, with #287 this would be guaranteed to not memcpy.

Comment (kristate) [reply std/tai64.zig]: It is used heavily as a wire format, so it's easy to create the TAI64N by simply grabbing it off the wire -- likewise for placing it back on the wire. The less CPU cycles the better.

Comment (kristate) [reply std/tai64.zig]: Also, I don't know if it is a wise idea to use it platform internal. to reiterate, TAI64N is a wire encoding format (think databases and packet headers).

Comment (andrewrk) [reply std/tai64.zig]: In general if you want to grab something off the wire and save cpu cycles, then you'll keep it encoded and never decode it. So you'll have a []u8 and we don't need to put it in a struct and give it a name. That's optimal and simpler. 

The reason you would want to put it in a struct and give it a name is to do something with the decoded information. For example, from_unixtime(), now(), to_unixtime() all deal with the decoded information. 

Equality comparison and ordering comparison can be done directly on the encoded bytes as you demonstrated, and as we do with utf8-encoded strings.

**Comment by andrewrk** [comment]
I'd like to see an example use case for this before accepting it into the standard library.

My recommendation is to continue working on whatever project it was where you needed this, and include the implementation in your project. Once you get to a certain milestone where we can see your project in action, and we can see how this TAI64 API is used, we can determine whether and how to upstream it into the zig standard library.

**Pull request closed without merging**